File: Templates/Project Templates/macOS/Application Extension/Audio Unit Extension.xctemplate/AudioUnitFactory.m
//___FILEHEADER___

#import "___PACKAGENAMEASIDENTIFIER___AudioUnit.h"
#import "AudioUnitFactory.h"


@implementation AudioUnitFactory

- (void)beginRequestWithExtensionContext:(NSExtensionContext *)context {

}

- (AUAudioUnit *)createAudioUnitWithComponentDescription:(AudioComponentDescription)desc error:(NSError **)error {
	return [[___PACKAGENAMEASIDENTIFIER___AudioUnit alloc] initWithComponentDescription:desc error:error];
}

@end

File: Templates/Project Templates/macOS/Application Extension/Audio Unit Extension.xctemplate/AudioUnitFactory.swift
//___FILEHEADER___

import CoreAudioKit

public class AudioUnitFactory: NSObject, AUAudioUnitFactory {
    var audioUnit: AUAudioUnit?

	public func beginRequest(with context: NSExtensionContext) {
	
	}

    @objc public func createAudioUnit(with componentDescription: AudioComponentDescription) throws -> AUAudioUnit {
        audioUnit = try ___PACKAGENAMEASIDENTIFIER___AudioUnit(componentDescription: componentDescription, options: [])
        
        return audioUnit!
    }
    
}